google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/stub/metrics/BigtableCloudMonitoringExporter.java

```diff
@@ -119,51 +162,148 @@ public CompletableResultCode export(Collection<MetricData> collection) {
     }
 
-    ProjectName projectName = ProjectName.of(projectId);
-    CreateTimeSeriesRequest request =
-        CreateTimeSeriesRequest.newBuilder()
-            .setName(projectName.toString())
-            .addAllTimeSeries(allTimeSeries)
-            .build();
+    ProjectName projectName = ProjectName.of(bigtableProjectId);
+    CreateTimeSeriesRequest bigtableRequest =
+            CreateTimeSeriesRequest.newBuilder()
+                    .setName(projectName.toString())
+                    .addAllTimeSeries(bigtableTimeSeries)
+                    .build();
 
-    ApiFuture<Empty> future = this.client.createServiceTimeSeriesCallable().futureCall(request);
+    ApiFuture<Empty> future =
+            this.client.createServiceTimeSeriesCallable().futureCall(bigtableRequest);
+
+    CompletableResultCode bigtableExportCode = new CompletableResultCode();
+    ApiFutures.addCallback(
+            future,
+            new ApiFutureCallback<Empty>() {
+              @Override
+              public void onFailure(Throwable throwable) {
+                logger.log(
+                        Level.WARNING,
+                        "createServiceTimeSeries request failed for bigtable metrics. ",
+                        throwable);
+                bigtableExportCode.fail();
+              }
 
-    lastExportCode = new CompletableResultCode();
+              @Override
+              public void onSuccess(Empty empty) {
+                bigtableExportCode.succeed();
+              }
+            },
+            MoreExecutors.directExecutor());
 
+    return bigtableExportCode;
+  }
+
+  /** Export metrics associated with the resource the Application is running on. */
+  private CompletableResultCode exportApplicationResourceMetrics(
+          Collection<MetricData> collection) {
+    if (applicationResource == null) {
+      return CompletableResultCode.ofSuccess();
+    }
+
+    // Filter application level metrics
+    List<MetricData> metricData =
+            collection.stream()
+                    .filter(md -> APPLICATION_METRICS.contains(md.getName()))
+                    .collect(Collectors.toList());
+
+    // Skip exporting if there's none
+    if (metricData.isEmpty()) {
+      return CompletableResultCode.ofSuccess();
+    }
+
+    List<TimeSeries> timeSeries;
+    try {
+      timeSeries =
+              BigtableExporterUtils.convertToApplicationResourceTimeSeries(
+                      metricData, taskId, applicationResource);
+    } catch (Throwable e) {
+      logger.log(
+              Level.WARNING,
+              "Failed to convert application metric data to cloud monitoring timeseries.",
+              e);
+      return CompletableResultCode.ofFailure();
+    }
+
+    // Construct the request. The project id will be the project id of the detected monitored
+    // resource.
+    ProjectName projectName =
+            ProjectName.of(applicationResource.getLabelsOrThrow(APPLICATION_RESOURCE_PROJECT_ID));
```

Comment:
is throwing ok here?
you took care to replace a thrown exception with a CompletableResultCode & logging above